django/db/backends/base/base.py

```diff
@@ -107,6 +107,13 @@ def __init__(self, settings_dict, alias=DEFAULT_DB_ALIAS):
         self.ops = self.ops_class(self)
         self.validation = self.validation_class(self)
 
+    def __repr__(self):
+        return '<%s settings_dict=%s... alias=%s>' % (
+            self.__class__.__qualname__,
+            repr(self.settings_dict)[:40],
```

Comment:
This can cause credential leaks on crash :fire: , e.g. in Sentry. I would leave only `alias` and `vendor`.